src/index.js

```diff
@@ -1,31 +1,36 @@
 // ITERATION 1
 
 function updateSubtotal(product) {
-  console.log('Calculating subtotal, yey!');
-
-  //... your code goes here
+ 
+
+  const price = product.querySelector('.price span').innerText;
+  const quantity = product.querySelector('.quantity input').value;
+  const subTotal = price * quantity
+  const returnSubtotal = product.querySelector('.subtotal span')
+  returnSubtotal.innerHTML = subTotal
+  return subTotal
+    
+    console.log('Calculating subtotal, yey!');
 }
 
 function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
-
-  // ITERATION 2
-  //... your code goes here
+  const products = document.getElementsByClassName('product');
 
-  // ITERATION 3
-  //... your code goes here
+  for(let i=0;i < products.length; i++){
+    updateSubtotal(products[i]);
```

Comment:
In this iteration you forgot to update the Total value, to do that you need to declare a variable where you will add the prices from all the products, and then you need to target the HTML tag for the total value and update it's value:

```
function calculateAll() {
  const products = document.getElementsByClassName("product");

  let sum = 0;

  for (let i = 0; i < products.length; i++) {
    updateSubtotal(products[i]);
    sum += updateSubtotal(products[i]);
  }

  let totalValue = document.querySelector("#total-value span");
  totalValue.innerHTML = sum;
}
```